taxi/views.py

```diff
@@ -87,3 +88,37 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    form_class = DriverCreationForm
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class AddCarToDriver(LoginRequiredMixin, generic.UpdateView):
+    model = Car
+    fields = ("drivers",)
+
+
+@login_required
+def add_cur_driver_to_car(request, pk: int):
+    Car.objects.get(id=pk).drivers.add(request.user)
```

Comment:
Replaces Car.objects.get(id=pk) with get_object_or_404(Car, id=pk). This ensures that if the car with the given ID does not exist, a 404 error page is shown instead of raising an unhandled exception.